server/routes/profile.js

```diff
@@ -0,0 +1,18 @@
+
+const express = require("express");
+const router = express.Router();
+const protect = require("../middleware/auth");
+const { searchProfiles,searchProfileById, updateProfile, createProfile } = require("../controllers/profile");
+
+router.route("/profiles").get(searchProfiles);
+
+router.route("/create_profile").post(protect, createProfile);
+
+router.route("/:id").get(searchProfileById); // im not sure if we are using this route to display profiles to a user once they are logged
+//in and see their own profile, or just to see other users profiles, its not clear to me, so if we want to display
+//all props of the profile to any user when they search, then we obviously do not want to send sensitive details of the persons profile
```

Comment:
We actually wouldn't necessarily be searching by the id directly but rather their name, location, etc. Of course, we don't have any advanced searching like Elasticsearch so our implementation will be simplified.

Reply:
But this route is still needed for fetching profile information on say our messages.

Reply:
So if we are going to make a Message schema with a reference to profile, I should keep this route as it is correct? We would probably search the profile by ID that is referenced in the Message document.

Reply:
Sure, this should be fine. We'll finalize the implementation in planning phases for the message feature. 👍🏻